linera-chain/src/chain.rs

```diff
@@ -169,6 +169,19 @@ where
     ViewError: From<C::Error>,
     C::Extra: ExecutionRuntimeContext,
 {
+    /// Substracts an amount from a balance and reports an error if that is impossible
+    fn sub_assign_fees(
+        balance: &mut Amount,
+        fees: Amount,
+        chain_execution_context: ChainExecutionContext,
+    ) -> Result<(), ChainError> {
+        let current_balance = *balance;
+        let error = SystemExecutionError::InsufficientFunding { current_balance };
+        let error = ExecutionError::SystemError(error);
+        let error = ChainError::ExecutionError(error, chain_execution_context);
```

Comment:
This code gets executed every time!

Reply:
I thought the LLVM would have handled it, but ok.